src/Data/Nat/DivMod.agda

```diff
@@ -223,6 +228,32 @@ m≥n⇒m/n>0 {m@(suc m-1)} {n@(suc n-1)} m≥n = begin
   n / d + (m * n) / d ≡⟨ cong (n / d +_) (*-/-assoc m d∣n) ⟩
   n / d + m * (n / d) ∎
 
+/-step : ∀ {m n n≢0} → m ≥ n → (m / n) {n≢0} ≡ 1 + ((m ∸ n) / n) {n≢0}
+/-step {zero} {zero} {()} _
+/-step {zero} {suc n} {_} ()
+/-step {suc m} {suc n} {n≢0} m≥n = begin-equality
+  suc m / suc n                          ≡⟨ refl ⟩
+  div-helper zero n (suc m) n            ≡⟨ divₕ-restart n (suc m) n m≥n ⟩
+  div-helper 1 n (suc m ∸ suc n) n       ≡⟨ divₕ-extractAcc 1 n (suc m ∸ suc n) n ⟩
+  1 + (div-helper 0 n (suc m ∸ suc n) n) ≡⟨ refl ⟩
+  1 + ((suc m ∸ suc n) / suc n)          ∎
+
+/-cancelˡ : ∀ m n o {o≢0} {mo≢0} → ((m * n) / (m * o)) {mo≢0} ≡ (n / o) {o≢0}
+/-cancelˡ m n o {o≢0} {mo≢0} = /-cancelˡ-Acc m (<-wellFounded n) o
+  where
+  /-cancelˡ-Acc : ∀ m {n} → (Acc _<_ n) → ∀ o {o≢0} {mo≢0} → ((m * n) / (m * o)) {mo≢0} ≡ (n / o) {o≢0}
+  /-cancelˡ-Acc (suc m) {n} (acc rec) o {o≢0} {mo≢0} with n <? o
```

Comment:
I'd use `m@(suc m-1)` instead of `(suc m)` so you can avoid writing `suc m` everywhere!